NUM_POINTS: 30
NUM_SETS: 20
MAX_NUM_POINTS_IN_SET: 10
